data/standard_features/14496-12/codecs.json

```diff
@@ -0,0 +1,279 @@
+    },
+    {
+      "fourcc": "resa",
+      "description": "Restricted audio",
+      "type": "AudioSampleEntry",
+      "containers": ["stsd"],
+      "syntax": "class AudioSampleEntry(codingname) extends SampleEntry (codingname){\r\n\tconst unsigned int(32)[2] reserved = 0;\r\n\tunsigned int(16) channelcount;\r\n\ttemplate unsigned int(16) samplesize = 16;\r\n\tunsigned int(16) pre_defined = 0;\r\n\tconst unsigned int(16) reserved = 0 ;\r\n\ttemplate unsigned int(32) samplerate = { default samplerate of media}<<16;\r\n\t// optional boxes follow\r\n\tBox ();\t\t// further boxes as needed\r\n\tChannelLayout();\r\n\tDownMixInstructions() [];\r\n\tDRCCoefficientsBasic() [];\r\n\tDRCInstructionsBasic() [];\r\n\tDRCCoefficientsUniDRC() [];\r\n\tDRCInstructionsUniDRC() [];\r\n\t// we permit only one DRC Extension box:\r\n\tUniDrcConfigExtension();\r\n\t// optional boxes follow\r\n\tSamplingRateBox();\r\n\tChannelLayout();\r\n}"
+    },
+    {
+      "fourcc": "resa",
+      "description": "Restricted audio",
+      "type": "AudioSampleEntryV1",
+      "containers": ["stsd"],
+      "syntax": "class AudioSampleEntryV1(codingname) extends SampleEntry (codingname){\r\n\tunsigned int(16) entry_version;\t// shall be 1, \r\n\t\t\t\t\t\t\t// and shall be in an stsd with version ==1\r\n\tconst unsigned int(16)[3] reserved = 0;\r\n\ttemplate unsigned int(16) channelcount;\t// shall be correct\r\n\ttemplate unsigned int(16) samplesize = 16;\r\n\tunsigned int(16) pre_defined = 0;\r\n\tconst unsigned int(16) reserved = 0 ;\r\n\ttemplate unsigned int(32) samplerate = 1<<16;\r\n\t// optional boxes follow\r\n\tSamplingRateBox();\r\n\tBox ();\t\t// further boxes as needed\r\n\tChannelLayout();\r\n\tDownMixInstructions() [];\r\n\tDRCCoefficientsBasic() [];\r\n\tDRCInstructionsBasic() [];\r\n\tDRCCoefficientsUniDRC() [];\r\n\tDRCInstructionsUniDRC() [];\r\n\t// we permit only one DRC Extension box:\r\n\tUniDrcConfigExtension();\r\n\t// optional boxes follow\r\n\tChannelLayout();\r\n}"
+    },
+    {
+      "fourcc": "resm",
+      "description": "Restricted metadata",
+      "type": "MetaDataSampleEntry",
+      "containers": ["stsd"],
+      "syntax": "class MetaDataSampleEntry(codingname) extends SampleEntry (codingname) {\r\n}"
+    },
+    {
+      "fourcc": "rest",
+      "description": "Restricted text",
+      "type": "SimpleTextSampleEntry",
+      "containers": ["stsd"],
+      "syntax": "TODO: Syntax from ISOBMFF Table 8 seems to refer to wrong SampleEntry class. It should be PlainTextSampleEntry"
```

Comment:
Leaving a comment on this to not forget about it

Reply:
Leaving a TODO for now.